Gu.Analyzers/Analyzers/ExceptionAnalyzer.cs

```diff
@@ -0,0 +1,37 @@
+namespace Gu.Analyzers.Analyzers
+{
+    using System;
+    using System.Collections.Immutable;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class ExceptionAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(GU0090DontThrowNotImplementedException.Descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ThrowExpression);
```

Comment:
We probably need to check other node types. IIRC `ThrowExpression` is for things like `?? throw ...`
The tests will reveal it, nice analyzer idea and the code looks nice.

Reply:
Yeah, I've switched it over to ``ThrowStatement``, Thanks.